### PR TITLE
Add detection for WPML WordPress plugin

### PR DESCRIPTION
**Pull request by strategio**
WPML is a very popular WordPress plugin to run multilingual site, more details on https://wpml.org/.

**Note:** There's already [WooCommerce Multilingual](https://github.com/wappalyzer/wappalyzer/blob/master/src/technologies/w.json#L1581) in the existing technologies. It was initially a glue plugin of WPML to run with WooCommerce, but since version 5.0 it also works standalone to offer multi-currency. So I didn't make WPML a requirement of WCML.

### CODE REVIEW

File: src/technologies/w.json [hunk 1]
```diff
@@ -1929,6 +1929,28 @@
...
+    "meta": {
+      "generator": "WPML"
+    },
+    "scriptSrc": "wp-content/plugins/sitepress-multilingual-cms",
+    "cookies": {
+      "wp-wpml_current_language": ""
+    },
+    "pricing": [
+      "low",
+      "recurring"
```
Comment (nurbek91) [inline src/technologies/w.json:1949]: onetime instead recurring maybe?

File: src/technologies/w.json [hunk 2]
```diff
@@ -1929,6 +1929,28 @@
...
+  "WPML": {
+    "cats": [
+      87,
+      89
+    ],
+    "description": "WPML plugin makes it possible to build and run fully multilingual WordPress sites.",
+    "icon": "WPML.svg",
+    "oss": true,
+    "meta": {
+      "generator": "WPML"
```
Comment (nurbek91) [inline src/technologies/w.json:1941]: `"generator": "^WPML\\sver\\:([\\d\\.]+)\\;version:\\1"`
examples:
https://www.avature.net/
https://www.tlscontact.com/
https://mixpanel.com/

File: src/technologies/w.json [hunk 3]
```diff
@@ -1929,6 +1929,28 @@
     ],
     "website": "https://www.wowza.com/video/player"
   },
+  "WPML": {
+    "cats": [
+      87,
+      89
+    ],
+    "description": "WPML plugin makes it possible to build and run fully multilingual WordPress sites.",
+    "icon": "WPML.svg",
```
Comment (nurbek91) [inline src/technologies/w.json:1938]: WPML
32x32

File: src/technologies/w.json [hunk 4]
```diff
@@ -1929,6 +1929,28 @@
...
+    "oss": true,
+    "meta": {
+      "generator": "WPML"
+    },
+    "scriptSrc": "wp-content/plugins/sitepress-multilingual-cms",
```
Comment (nurbek91) [inline src/technologies/w.json:1943]: `"scriptSrc": "/wp-content/plugins/sitepress-multilingual-cms/.+\\.js(?:\\?ver=(\\d+(?:\\.\\d+)+))?\\;version:\\1",`

**Review by nurbek91 — COMMENTED** [review]
Hey mate, thanks for PR!!

File: src/technologies/w.json (hunk 1 above)
Comment (strategio) [reply src/technologies/w.json:1949]: I assume this is correct, WPML is sold as a 1 year licence automatically renewable.

File: src/technologies/w.json (hunk 2 above)
Comment (strategio) [reply src/technologies/w.json]: Thanks @nurbek91, adjusted!

File: src/technologies/w.json (hunk 4 above)
Comment (strategio) [reply src/technologies/w.json]: @nurbek91, I did some tests and it's not working (I mean we can't get the version number from that). Also tried other patterns, unsuccessfully.

The problem is that we have "template" scripts with their own version number (not related to the plugin's version).
We actually may have 1 WPML script loaded on the frontend, but it depends on the site settings.

I prefer not trying to get the version from this rule.

**Comment by strategio** [comment]
@nurbek91, thanks for you review!

FYI, I think all rules are "confident", but some are more likely to be found. Indeed, the script and the cookie might be available but it also depends on the site setup (depends on 2 different settings).

The rule that might be the most efficient for detection is the meta generator.

Let me know if you have other comments.

File: src/technologies/w.json (hunk 1 above)
Comment (nurbek91) [reply src/technologies/w.json:1949]: ok mate

**Review by nurbek91 — APPROVED** [review]

**Review by nurbek91 — APPROVED** [review]

**Comment by nurbek91** [comment]
> @nurbek91, thanks for you review!
> 
> FYI, I think all rules are "confident", but some are more likely to be found. Indeed, the script and the cookie might be available but it also depends on the site setup (depends on 2 different settings).
> 
> The rule that might be the most efficient for detection is the meta generator.
> 
> Let me know if you have other comments.

Hey mate, everything looks great. Thanks for PR!!

**Comment by strategio** [comment]
Hey @nurbek91, will you merge it? Or do need anything else from me? Thanks!

**Comment by nurbek91** [comment]
> Hey @nurbek91, will you merge it? Or do need anything else from me? Thanks!



> Hey @nurbek91, will you merge it? Or do need anything else from me? Thanks!

Elbert will merge it. Approximately at the beginning of the each week.

**Comment by strategio** [comment]
Thanks guys!

**Pull request merged**